ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+ENABLE_AUTOSCALING=${ENABLE_AUTOSCALING:-true}
+REPLICAS=${REPLICAS:-3}
+MIN_REPLICAS=${MIN_REPLICAS:-2}
+# wait for all nodes to reach Ready=true to ensure that all machines and nodes came up, before we run
+# any e2e tests that might require specific workload capacity.
+echo "$(date) - waiting for worker nodes to be ready..."
+ret=0
+oc wait nodes --all --for=condition=Ready=true --timeout=10m || ret=$?
+if [[ "$ret" == 0 ]]; then
+    NODECOUNT="$(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/SchedulingDisabled!=,node-role.kubernetes.io/infra!=,node-role.kubernetes.io/workload!= | wc -l)"
+    if [[ "$ENABLE_AUTOSCALING" == "false" ]]; then
+        # Check worker node count matches requested REPLICAS
+        if [[ "$NODECOUNT" == "$REPLICAS" ]]; then
+            echo "$(date) - all $NODECOUNT nodes are ready and match requested $REPLICAS replica count"
+        else
+            echo "Requested worker nodes: $REPLICAS"
+            echo "Worker nodes in Ready state: $NODECOUNT"
+            oc get nodes
+            echo "$(date) - Node count does not match requested replica count. Exiting..."
+            exit 1
+        fi
+    else
+        # Check and confirm node count is no less than MIN_REPLICAS.
+        if (( "$NODECOUNT" >= "$MIN_REPLICAS" )); then
+            echo "All nodes are ready"
+        else
+            echo "Requested minimum worker nodes: $MIN_REPLICAS"
+            echo "Worker nodes in Ready state: $NODECOUNT"
+            oc get nodes
+            echo "Nodes did not become ready. Exiting.."
+            exit 1
+        fi
+    fi
```

Comment:
As we get `desired_node_count` in the previous step, `"$node_count" >= "$desired_node_count"` is enough for the validation. The step `rosa-nodes-wait-ready` is a helping step for testing, but not a feature validation step.